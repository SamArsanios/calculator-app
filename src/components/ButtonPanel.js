import React from 'react';
import PropTypes from 'prop-types';
import Button from './Button';

export default ButtonPanel = () => {
    return (
        <div className="button-panel">
            <div className="">
                <Button buttonName />
                <Button buttonName />
                <Button buttonName />
            </div>
        </div>
    );
}