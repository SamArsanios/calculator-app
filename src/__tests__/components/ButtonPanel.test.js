import React from 'react';
import renderer from 'react-test-renderer';
import ButtonPanel from '../../components/ButtonPanel';

const test = () => { };

describe('ButtonPanel', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<ButtonPanel name="1" clickHandler={test} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});