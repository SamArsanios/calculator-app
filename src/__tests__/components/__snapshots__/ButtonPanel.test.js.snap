// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Panel renders correctly 1`] = `
<div
  className="button-panel"
>
  <div
    className="buttons font"
  >
    <button
      className="button grey "
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="button grey "
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="button grey "
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="button orange "
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
  </div>
  <div
    className="buttons font"
  >
    <button
      className="button grey "
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="button grey "
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="button grey "
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="button orange "
      onClick={[Function]}
      type="button"
    >
      *
    </button>
  </div>
  <div
    className="buttons"
  >
    <button
      className="button grey "
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="button grey "
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="button grey "
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="button orange "
      onClick={[Function]}
      type="button"
    >
      -
    </button>
  </div>
  <div
    className="buttons"
  >
    <button
      className="button grey "
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="button grey "
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="button grey "
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="button orange "
      onClick={[Function]}
      type="button"
    >
      +
    </button>
  </div>
  <div
    className="buttons"
  >
    <button
      className="button grey width"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="button grey "
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="button orange "
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;

exports[`Button renders correctly 1`] = `
<div
  className="button-panel"
>
  <div
    className="buttons font"
  >
    <button
      className="button grey "
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="button grey "
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="button grey "
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="button orange "
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
  </div>
  <div
    className="buttons font"
  >
    <button
      className="button grey "
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="button grey "
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="button grey "
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="button orange "
      onClick={[Function]}
      type="button"
    >
      *
    </button>
  </div>
  <div
    className="buttons"
  >
    <button
      className="button grey "
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="button grey "
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="button grey "
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="button orange "
      onClick={[Function]}
      type="button"
    >
      -
    </button>
  </div>
  <div
    className="buttons"
  >
    <button
      className="button grey "
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="button grey "
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="button grey "
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="button orange "
      onClick={[Function]}
      type="button"
    >
      +
    </button>
  </div>
  <div
    className="buttons"
  >
    <button
      className="button grey width"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="button grey "
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="button orange "
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;

exports[`ButtonPanel renders correctly 1`] = `
<div
  className="button-panel"
>
  <div
    className="buttons font"
  >
    <button
      className="button grey "
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="button grey "
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="button grey "
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="button orange "
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
  </div>
  <div
    className="buttons font"
  >
    <button
      className="button grey "
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="button grey "
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="button grey "
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="button orange "
      onClick={[Function]}
      type="button"
    >
      *
    </button>
  </div>
  <div
    className="buttons"
  >
    <button
      className="button grey "
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="button grey "
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="button grey "
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="button orange "
      onClick={[Function]}
      type="button"
    >
      -
    </button>
  </div>
  <div
    className="buttons"
  >
    <button
      className="button grey "
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="button grey "
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="button grey "
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="button orange "
      onClick={[Function]}
      type="button"
    >
      +
    </button>
  </div>
  <div
    className="buttons"
  >
    <button
      className="button grey width"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="button grey "
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="button orange "
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
